### PR TITLE
MF-1636 - Create tests for certs endpoint

### LINKED ISSUE #1636: Create tests for certs endpoints
**FEATURE REQUEST**

Certificate endpoints needs unit tests

### PR DESCRIPTION
**Pull request by zzokki81**
Signed-off-by: zzokki81 <zoran.rebic@outlook.com>

### What does this do?
Added tests for certs endpoints
### Which issue(s) does this PR fix/relate to?
Resolves #1636

### List any changes that modify/break current functionality
N/A
### Have you included tests for your changes?
Certs endpoint tests
### Did you document any new/modified functionality?
No
### Notes
N/A

### CODE REVIEW

File: certs/api/endpoint_test.go [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+	require.Nil(t, err, fmt.Sprintf("unexpected service creation error: %s\n", err))
+
+	cs := newCertServer(svc)
+	serialURL := fmt.Sprintf("%s/serials/%s", cs.URL, thingID)
+	defer cs.Close()
+
+	var issuedCerts []certsRes
+	for i := 0; i < certNum; i++ {
+		cert, err := svc.IssueCert(context.Background(), token, thingID, ttl, keyBits, key)
+		require.Nil(t, err, fmt.Sprintf("Saving config expected to succeed: %s.\n", err))
```
Comment (manuio) [inline certs/api/endpoint_test.go]: I'd say `Certificate issuing` instead of `Saving config`

File: certs/api/endpoint_test.go [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+		},
+		{
+			desc:        "issue new cert with wrong JSON",
+			req:         "{\"thing_id\": 5}",
+			auth:        token,
+			contentType: contentType,
+			status:      http.StatusInternalServerError,
+		},
+		{
+			desc:        "issue new cert  with invalid request format",
```
Comment (manuio) [inline certs/api/endpoint_test.go]: Fix the double white space

File: certs/api/endpoint_test.go [hunk 3]
```diff
@@ -0,0 +1,545 @@
...
+		},
+		{
+			desc:   "list all certs with limit greater then allowed",
+			url:    fmt.Sprintf("%s?offset=%d&limit=%d", serialURL, 0, 1000),
+			auth:   token,
+			certs:  certsPageRes{},
+			status: http.StatusBadRequest,
+		},
+		{
+			desc:   "list all certs with invalid limit ",
```
Comment (manuio) [inline certs/api/endpoint_test.go]: Remove the white space at the end of the sentence

File: certs/api/endpoint_test.go [hunk 4]
```diff
@@ -0,0 +1,545 @@
...
+			status: http.StatusUnauthorized,
+		},
+		{
+			desc:   "list certificate data with empty token",
+			id:     saved.Serial,
+			auth:   "",
+			status: http.StatusUnauthorized,
+		},
+		{
+			desc:   "list certificate data with wrong certID",
```
Comment (manuio) [inline certs/api/endpoint_test.go]: `with invalid certificate ID`

File: certs/api/endpoint_test.go [hunk 5]
```diff
@@ -0,0 +1,545 @@
...
+			id:     wrongCertID,
+			auth:   token,
+			status: http.StatusInternalServerError,
+		},
+		{
+			desc:   "list certificate data with no certID",
```
Comment (manuio) [inline certs/api/endpoint_test.go]: `with empty cerificatet ID`

File: certs/api/endpoint_test.go [hunk 6]
```diff
@@ -0,0 +1,545 @@
...
+	}{
+		{
+			desc:   "list all certs ID's",
+			url:    serialURL,
+			auth:   token,
+			certs:  certsPageRes{pageRes: pageRes{Total: uint64(certNum), Offset: 0, Limit: certNum}, Certs: issuedCerts},
+			status: http.StatusOK,
+		},
+		{
+			desc:   "list certs serial ID's for wrong thing",
```
Comment (manuio) [inline certs/api/endpoint_test.go]: `with invalid thing ID`

File: certs/api/endpoint_test.go [hunk 7]
```diff
@@ -0,0 +1,545 @@
...
+
+func TestIssueCert(t *testing.T) {
+	svc, err := newCertService(map[string]string{token: email})
+	require.Nil(t, err, fmt.Sprintf("unexpected service creation error: %s\n", err))
+
+	cs := newCertServer(svc)
+	defer cs.Close()
+
+	data := toJSON(addReq)
+	neThingID := addReq
```
Comment (manuio) [inline certs/api/endpoint_test.go]: I'd use `wrongAddReq`

File: certs/api/endpoint_test.go [hunk 8]
```diff
@@ -0,0 +1,545 @@
...
+	}{
+		{
+			desc:        "issue new cert",
+			req:         data,
+			auth:        token,
+			contentType: contentType,
+			status:      http.StatusCreated,
+		},
+		{
+			desc:        "issue new cert for non existing thing id",
```
Comment (manuio) [inline certs/api/endpoint_test.go]: `thing ID`

**Review by manuio — CHANGES_REQUESTED** [review]

File: certs/api/endpoint_test.go [hunk 9]
```diff
@@ -407,13 +407,13 @@ func TestViewCert(t *testing.T) {
...
 		{
-			desc:   "list certificate data with wrong certID",
+			desc:   "list certificate data with invalid certificate ID",
 			id:     wrongCertID,
 			auth:   token,
 			status: http.StatusInternalServerError,
 		},
 		{
-			desc:   "list certificate data with no certID",
+			desc:   "list certificate data with empty certID",
```
Comment (manuio) [inline certs/api/endpoint_test.go]: `empty certificate ID`

**Review by manuio — CHANGES_REQUESTED** [review]

**Comment by dborovcanin** [comment]
This PR has some interesting ideas, but it's been out of date with no signs of any updates, so I'm closing it. @nyagamunene Please take a look and consider this for MG Certs service tests in a different PR.

**Pull request closed without merging**